src/prefect/core/task.py

```diff
@@ -671,7 +671,11 @@ def bind(
 
         flow = flow or prefect.context.get("flow", None)
         if not flow:
-            raise ValueError("Could not infer an active Flow context.")
+            raise ValueError(
+                f"Could not infer an active Flow context while creating edge to {self}."
+                " This often means you called a task outside a `with Flow(...)` block. "
+                "Did you mean to call `this_task.run(...)`?"
```

Comment:
We could say `{self.__name__}.run(...)` but if their task is a class e.g. `FooBar` it will be misleading because it will not indicate it's instantiated / we can't know the name of their variable. Perhaps we should optimize for people using `@task` decorators instead since that would be a more helpful message.